app/main.py

```diff
@@ -39,51 +48,257 @@ def execute(self):
 
             self.execute_as_loop(30)  # 30-second interval.
         """
+        self.load_shelve()
 
     def shutdown(self):
         """Run when your NApp is unloaded.
 
         If you have some cleanup procedure, insert it here.
         """
 
+    @listen_to("kytos/topology.unloaded")
+    def unload_topology(self):  # pylint: disable=W0613
+        """Function meant for validation, to make sure that the shelve
+        has been loaded before all the other functions that use it begins to
+        call it."""
+        self.shelve_loaded = False  # pylint: disable=W0201
+
+    def test_kytos_topology(self):
```

Comment:
This method doesn't seem to be used anywhere. If you could double check.

Reply:
I will remove it since it is from outdated code.

Reply:
Looks good